Library/Homebrew/cask/lib/hbc/artifact/pkg.rb

```diff
@@ -52,7 +54,25 @@ def run_installer(pkg_description)
         ]
         args << "-verboseR" if Hbc.verbose
         args << "-allowUntrusted" if pkg_install_opts :allow_untrusted
-        @command.run!("/usr/sbin/installer", sudo: true, args: args, print_stdout: true)
+        with_choices_file pkg_install_opts(:choices) do |choices_path|
+          args << "-applyChoiceChangesXML" << choices_path if choices_path
+          @command.run!("/usr/sbin/installer", sudo: true, args: args, print_stdout: true)
+        end
+      end
+
+      def with_choices_file(choices)
+        unless choices
+          yield nil
+          return
+        end
+
+        begin
+          file = Tempfile.new(["choices", ".xml"])
+          file.write Plist::Emit.dump(choices)
+          yield file.path
+        ensure
+          file.close(true)
+        end
```

Comment:
I'm afraid I don't understand your point.

`::open` with a block does not delete the file but the finalizer does.
This behaviour is essentially same as `::new`; both delete no files until Ruby's GC runs.
The difference between them is `::open` closes the IO handle, which is also eventually closed with `::new` with GC.
I am sure ensuring the temp file deleted needs to call `#close(true)` or `#unlink` explicitly, or use `::create`, which is unavailable at 2.0.0.

See http://ruby-doc.org/stdlib-2.0.0/libdoc/tempfile/rdoc/Tempfile.html#method-c-open

I wish I did not miss any major points.